generate_side_by_side.py

```diff
@@ -355,6 +368,118 @@ def _open_data(file):
         most_similar=most_similar,
     )
 
+def generate_step_chart(oldvid, newvid, vismetPath, prefix, metric):
+  oldvid_metrics = json.loads(
+    subprocess.check_output(
+      [
+          "python",
+          vismetPath,
+          "--orange",
+          "--perceptual",
+          "--contentful",
+          "--force",
+          "--renderignore",
+          "5",
+          "--json",
+          "--viewport",
+          "--video",
+          oldvid
+      ]
+    )
+  )
+
+  newvid_metrics = json.loads(
+    subprocess.check_output(
+      [
+          "python",
+          vismetPath,
+          "--orange",
+          "--perceptual",
+          "--contentful",
+          "--force",
+          "--renderignore",
+          "5",
+          "--json",
+          "--viewport",
+          "--video",
+          newvid
+      ]
+    )
+  )
+
+  if metric.lower() == "perceptualspeedinex":
+    progress="PerceptualSpeedIndexProgress"
+    metricName="PerceptualSpeedIndex"
+  elif metric.lower() == "contentfulspeedindex":
+    progress="ContentfulSpeedIndexProgress"
+    metricName="ContentfulSpeedIndex"
+  else:
+    progress="VisualProgress"
+    metricName="SpeedIndex"
+
+  x = []
+  y = []
+  for pt in oldvid_metrics[progress].split(","):
+    x.append(int(pt.split("=")[0]))
+    y.append(int(pt.split("=")[1]))
+  plt.step(x, y, label="Before (%d)"% oldvid_metrics[metricName])
+
+
+  x = []
+  y = []
+  for pt in newvid_metrics[progress].split(","):
+    x.append(int(pt.split("=")[0]))
+    y.append(int(pt.split("=")[1]))
+  plt.step(x, y, label="After (%d)"% newvid_metrics[metricName])
+
+  plt.legend(loc="lower right")
+  plt.title("%s %s"% (prefix.rstrip("_"), metricName))
+  plt.savefig("%s-%s-step.png"% (prefix.rstrip("_"), metric))
+  plt.clf()
+
+
+def find_closest_videos(base_videos, base_vismet, new_videos, new_vismet, output, prefix, metric):
+
+  for m in base_vismet:
+    if metric.lower() == m["name"].lower():
+      diff = [abs(m["replicates"][i]-m["value"]) for i in range(0,len(m["replicates"]))]
+      base_min_diff = min(diff)
```

Comment:
Rather than making two passes in the diff list to find the min and the index you can use np.argmin to return the index and then you can access the element: https://numpy.org/doc/stable/reference/generated/numpy.argmin.html
```
base_min_ind = np.argmin(diff)
```